### PR TITLE
Refactor the Metric Application Layer

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Fixes https://github.com/SigNoz/signoz/issues/2974

 - Removed the unnecessary  dependency on the redux for storing metrics
 - Removed the API dependency for fetching top-operation, service-overview, and top-level-operation.

### CODE REVIEW

File: frontend/src/api/ErrorResponseHandler.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export function ErrorResponseHandler(error: AxiosError): ErrorResponse {
 				return {
 					statusCode,
 					payload: null,
-					error: data.errorType,
+					error: data,
```
Comment (palashgdev) [inline frontend/src/api/ErrorResponseHandler.ts:19]: why this change is here

File: frontend/src/api/metrics/getServiceOverview.ts [hunk 2]
```diff
@@ -1,30 +1,16 @@
...
-): Promise<SuccessResponse<PayloadProps> | ErrorResponse> => {
-	try {
-		const response = await axios.post(`/service/overview`, {
-			start: `${props.start}`,
-			end: `${props.end}`,
-			service: props.service,
-			step: props.step,
-			tags: props.selectedTags,
-		});
+const getServiceOverview = async (props: Props): Promise<PayloadProps> => {
```
Comment (palashgdev) [inline frontend/src/api/metrics/getServiceOverview.ts:4]: can we have Promise<SuccessResponse<PayloadProps> | ErrorResponse>

File: frontend/src/api/metrics/getTopLevelOperations.ts [hunk 3]
```diff
@@ -1,24 +1,9 @@
 import axios from 'api';
-import { ErrorResponseHandler } from 'api/ErrorResponseHandler';
-import { AxiosError } from 'axios';
-import { ErrorResponse, SuccessResponse } from 'types/api';
 import { PayloadProps, Props } from 'types/api/metrics/getTopLevelOperations';
 
-const getTopLevelOperations = async (
-	props: Props,
-): Promise<SuccessResponse<PayloadProps> | ErrorResponse> => {
```
Comment (palashgdev) [inline frontend/src/api/metrics/getTopLevelOperations.ts:9]: same here also

File: frontend/src/api/metrics/getTopLevelOperations.ts [hunk 4]
```diff
@@ -1,24 +1,9 @@
...
-			error: null,
-			message: response.data.status,
-			payload: response.data[props.service],
-		};
-	} catch (error) {
-		return ErrorResponseHandler(error as AxiosError);
-	}
+const getTopLevelOperations = async (props: Props): Promise<PayloadProps> => {
+	const response = await axios.post(`/service/top_level_operations`);
+	return response.data[props.service];
```
Comment (palashgdev) [inline frontend/src/api/metrics/getTopLevelOperations.ts:6]: null handling

Comment (palashgdev) [inline frontend/src/api/metrics/getTopLevelOperations.ts:6]: also no logic should be written in this service

File: frontend/src/api/metrics/getTopOperations.ts [hunk 5]
```diff
@@ -1,29 +1,15 @@
...
-	props: Props,
-): Promise<SuccessResponse<PayloadProps> | ErrorResponse> => {
-	try {
-		const response = await axios.post(`/service/top_operations`, {
-			start: `${props.start}`,
-			end: `${props.end}`,
-			service: props.service,
-			tags: props.selectedTags,
-		});
+const getTopOperations = async (props: Props): Promise<PayloadProps> => {
```
Comment (palashgdev) [inline frontend/src/api/metrics/getTopOperations.ts:4]: same here

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 6]
```diff
@@ -37,9 +47,26 @@ import {
 } from './util';
 
 function Application({ getWidgetQueryBuilder }: DashboardProps): JSX.Element {
+	const globalTime = useSelector<AppState, GlobalReducer>(
+		(state) => state.globalTime,
+	);
+
+	const { maxTime, minTime } = GetMinMax(globalTime.selectedTime, [
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:54]: we already have minMax time from globalTime reducer no need for this one

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 7]
```diff
@@ -64,12 +91,96 @@ function Application({ getWidgetQueryBuilder }: DashboardProps): JSX.Element {
...
+			globalTime.minTime,
+			globalTime.maxTime,
+		],
+		() =>
+			getTopLevelOperations({
+				service: servicename || '',
+			}),
+	);
+
+	useEffect(() => {
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:163]: no need for this useEffect we have isError from the useQuery which will give you this data

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 8]
```diff
@@ -64,12 +91,96 @@ function Application({ getWidgetQueryBuilder }: DashboardProps): JSX.Element {
...
+				selectedTags,
+			}),
+	);
+
+	const {
+		data: topLevelOperations,
+		error: topLevelOperationsError,
+		isError: topLevelOperationsIsError,
+		isLoading: topLevelOperationsLoading,
+	} = useQuery(
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:149]: use useQueries when we are working with multiple queries

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 9]
```diff
@@ -64,12 +91,96 @@ function Application({ getWidgetQueryBuilder }: DashboardProps): JSX.Element {
...
 
-	const { queries } = useResourceAttribute();
+	const {
+		data: topOperations,
+		error: topOperationsError,
+		isError: topOperationsIsError,
+		isLoading: topOperationsLoading,
+	} = useQuery(
+		[
+			`topOperation`,
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:129]: no magic string

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 10]
```diff
@@ -64,12 +91,96 @@ function Application({ getWidgetQueryBuilder }: DashboardProps): JSX.Element {
...
+	const {
+		data: serviceOverview,
+		error: serviceOverviewError,
+		isError: serviceOverviewIsError,
+		isLoading: serviceOverviewIsLoading,
+	} = useQuery(
+		[
+			`getServiceOverview`,
+			servicename,
+			getStep({ start: minTime, end: maxTime, inputFormat: 'ns' }),
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:103]: minMax time are sufficient to have key no need for the function call here

File: frontend/src/lib/dashbaordVariables/getDashboardVariables.ts [hunk 11]
```diff
@@ -7,24 +7,28 @@ export const getDashboardVariables = (): Record<string, unknown> => {
...
-		const [selectedDashboard] = dashboards;
-		const {
-			data: { variables = {} },
-		} = selectedDashboard;
 
-		const { start, end } = getStartEndRangeTime({
-			type: 'GLOBAL_TIME',
-			interval: globalTime.selectedTime,
-		});
+		if (dashboards.length > 0) {
```
Comment (palashgdev) [inline frontend/src/lib/dashbaordVariables/getDashboardVariables.ts:11]: this change is not relevant please remove this change

File: frontend/src/pages/MetricApplication/index.tsx [hunk 12]
```diff
@@ -1,76 +1,7 @@
...
-	if (metricsApplicationLoading) {
-		return <Spinner tip="Loading..." />;
-	}
-
-	if (error) {
-		return <Typography>{errorMessage}</Typography>;
-	}
 
+function MetricsApplication(): JSX.Element {
 	return <MetricsApplicationContainer />;
```
Comment (palashgdev) [inline frontend/src/pages/MetricApplication/index.tsx:4]: if we are returning only one single component’s no sense writing here

**Review by palashgdev — CHANGES_REQUESTED** [review]
done the code review need to check from functionality perspective

**Comment by palashgdev** [comment]
also build pipeline is breaking

File: frontend/src/api/ErrorResponseHandler.ts (hunk 1 above)
Comment (Rajat-Dabade) [reply frontend/src/api/ErrorResponseHandler.ts:19]: There is no `data.errorType` in the response object of AxiosError.

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 6 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:54]: Okay

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 7 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:163]: Okay, I will remove it.

File: frontend/src/api/ErrorResponseHandler.ts (hunk 1 above)
Comment (palashgdev) [reply frontend/src/api/ErrorResponseHandler.ts:19]: let's revert this change though it used everywhere

File: frontend/src/api/metrics/getTopLevelOperations.ts (hunk 4 above)
Comment (Rajat-Dabade) [reply frontend/src/api/metrics/getTopLevelOperations.ts:6]: Sure will do it.

Comment (Rajat-Dabade) [reply frontend/src/api/metrics/getTopLevelOperations.ts:6]: Sure, I will remove it.

File: frontend/src/lib/dashbaordVariables/getDashboardVariables.ts (hunk 11 above)
Comment (Rajat-Dabade) [reply frontend/src/lib/dashbaordVariables/getDashboardVariables.ts:11]: Done

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 10 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx]: Okay

File: frontend/src/pages/MetricApplication/index.tsx (hunk 12 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/MetricApplication/index.tsx:4]: As of now, we don't have any logic here, if we have to move we have to move some container login which I think will not be a great idea to do.  What I think is let's keep the pages and it's logic in `/pages/MetricApplication/index.tsx` part in a container and it's logic in `container/MetricApplication`

Thought?

**Comment by palashgdev** [comment]
@Rajat-Dabade can you check 

```
TypeError: Cannot destructure property 'data' of 'selectedDashboard' as it is undefined.
```

**Comment by Rajat-Dabade** [comment]
@palashgdev, this was because of this issue https://github.com/SigNoz/signoz/issues/2996.

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 13]
```diff
@@ -64,12 +84,59 @@ function Application(): JSX.Element {
...
+				selectedTags,
+			}),
+	};
+
+	const topLevelOperationsQuery = {
+		queryKey: [servicename, minTime, maxTime, selectedTags],
+		queryFn: (): Promise<ServiceDataProps> => getTopLevelOperations(),
+	};
+
+	const queryResult: UseQueryResult[] = useQueries([
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:119]: utilize the genericity of UseQueryResult so that we don't have to enforce types

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 14]
```diff
@@ -237,17 +332,26 @@ function Application(): JSX.Element {
...
-								fullViewOptions={false}
-								onClickHandler={handleGraphClick('Rate')}
-								widget={operationPerSecWidget}
-								yAxisUnit="ops"
-								onDragSelect={onDragSelect}
-							/>
-						</GraphContainer>
+						{topLevelOperationsIsError && (
+							<Typography>
+								{(topLevelOperationsError as AxiosError).response?.data}
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:337]: there is a method axios.isAxiosError

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 15]
```diff
@@ -64,12 +84,59 @@ function Application(): JSX.Element {
...
+		topOperationsQuery,
+		topLevelOperationsQuery,
+	]);
+
+	const serviceOverview = queryResult[0].data as PayloadProps;
+	const serviceOverviewError = queryResult[0].error;
+	const serviceOverviewIsError = queryResult[0].isError;
+	const serviceOverviewIsLoading = queryResult[0].isLoading;
+
+	const topOperations = queryResult[1].data as PayloadPropsTopOpertions;
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:130]: make this as component so you don't need to to create for every component

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/types/api/metrics/getTopLevelOperations.ts [hunk 16]
```diff
@@ -5,3 +5,7 @@ export interface Props {
 }
 
 export type PayloadProps = TopLevelOperations;
+
+export type ServiceDataProps = {
```
Comment (palashgdev) [inline frontend/src/types/api/metrics/getTopLevelOperations.ts:9]: remove this type from here

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Comment by palashgdev** [comment]
> function MetricsApplication(): JSX.Element {

this is resolved as this was the issue on my end

File: frontend/src/types/api/metrics/getTopLevelOperations.ts (hunk 16 above)
Comment (Rajat-Dabade) [reply frontend/src/types/api/metrics/getTopLevelOperations.ts:9]: Sure

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 14 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:337]: axios.isAxiosError is throwing Error `Cannot read the property of undefined`, better to keep it this way.

Thought?

File: frontend/src/pages/MetricApplication/index.tsx (hunk 12 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/MetricApplication/index.tsx]: Move the logic from container layer to pages layer

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 17]
```diff
@@ -64,12 +83,59 @@ function Application(): JSX.Element {
 		[handleSetTimeStamp],
 	);
 
-	const { topOperations, serviceOverview, topLevelOperations } = useSelector<
-		AppState,
-		MetricReducer
-	>((state) => state.metrics);
+	const serviceOverviewQuery = {
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:86]: also no need of this object let use this one directly

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 18]
```diff
@@ -64,12 +83,59 @@ function Application(): JSX.Element {
...
+				step: getStep({
+					start: minTime,
+					end: maxTime,
+					inputFormat: 'ns',
+				}),
+				selectedTags,
+			}),
+	};
+
+	const topOperationsQuery = {
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:102]: don't create a constant use directly

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 14 above)
Comment (palashgdev) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:336]: this is about enforcing about types

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 19]
```diff
@@ -64,12 +83,59 @@ function Application(): JSX.Element {
...
+			}),
+	};
 
-	const { queries } = useResourceAttribute();
+	const topLevelOperationsQuery = {
+		queryKey: [servicename, minTime, maxTime, selectedTags],
+		queryFn: (): Promise<ServiceDataProps> => getTopLevelOperations(),
+	};
+
+	const queryResult = useQueries([
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:118]: use type here for query's as you don't need to enforce the type

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 14 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:336]: Okay

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 17 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:86]: Okay

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 18 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:102]: okay

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 19 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:118]: Okay

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 15 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:129]: This can be done after merging of https://github.com/SigNoz/signoz/pull/3107

File: frontend/src/store/reducers/global.ts [hunk 20]
```diff
@@ -12,7 +12,7 @@ import {
 } from 'types/actions/logs';
 import { GlobalReducer } from 'types/reducer/globalTime';
 
-const intitalState: GlobalReducer = {
+export const initialState: GlobalReducer = {
```
Comment (palashgdev) [inline frontend/src/store/reducers/global.ts:15]: let's revert this change

File: frontend/src/pages/MetricApplication/index.tsx [hunk 21]
```diff
@@ -1,76 +1,89 @@
...
 
-	if (error) {
-		return <Typography>{errorMessage}</Typography>;
-	}
+	const searchParams = new URLSearchParams(search);
+	const tab = searchParams.get('tab');
 
-	return <MetricsApplicationContainer />;
-}
+	const overMetrics = 'Overview Metrics';
```
Comment (palashgdev) [inline frontend/src/pages/MetricApplication/index.tsx:31]: make a constant

File: frontend/src/pages/MetricApplication/index.tsx [hunk 22]
```diff
@@ -1,76 +1,89 @@
...
+	const dbCallMetrics = 'Database Calls';
+	const externalMetrics = 'External Calls';
 
-interface DispatchProps {
-	getInitialData: (props: GetInitialDataProps) => void;
-}
+	const getActiveKey = (): string => {
```
Comment (palashgdev) [inline frontend/src/pages/MetricApplication/index.tsx:35]: const getActiveKey = (): string => {
    const metricsMap = {
        [dbCallMetrics]: dbCallMetrics,
        [externalMetrics]: externalMetrics,
    };

    return metricsMap[tab] || overMetrics;
};

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 23]
```diff
@@ -265,23 +366,41 @@ function Application(): JSX.Element {
...
-								fullViewOptions={false}
-								onClickHandler={handleGraphClick('Error')}
-								widget={errorPercentageWidget}
-								yAxisUnit="%"
-								onDragSelect={onDragSelect}
-							/>
-						</GraphContainer>
+						{topLevelOperationsIsError && (
+							<Typography>
+								{axios.isAxiosError(topLevelOperationsError) &&
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:371]: what if this is not axios error ?

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx [hunk 24]
```diff
@@ -237,17 +329,26 @@ function Application(): JSX.Element {
...
-								yAxisUnit="ops"
-								onDragSelect={onDragSelect}
-							/>
-						</GraphContainer>
+						{topLevelOperationsIsError && (
+							<Typography>
+								{(topLevelOperationsError as AxiosError).response?.data}
+							</Typography>
+						)}
+						{!topLevelOperationsIsError && !topLevelOperationsLoading && (
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/Overview.tsx:337]: handling this one with enabled

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/pages/MetricApplication/index.tsx (hunk 22 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/MetricApplication/index.tsx:35]: But the `tab` can be null, then it will be like 

```ts
const getActiveKey = (): string => {
	const metricsMap = {
		[dbCallMetrics]: dbCallMetrics,
		[externalMetrics]: externalMetrics
	}
	if (tab) {
		return metricsMap[tab] || overMetrics
	} else {
		return overMetrics
	}
}
```

Will this be better, performance vise both approaches have similar performance in terms of execution time.

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 23 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:371]: Will set the Something went wrong message.

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 24 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx:337]: Here `enabled` will not work because `isLoading` will only be true when an API request will be sent, enabled will not send the API request in the first place so `isLoading` will be `false` always. I have handled it in another way. Please check the functionality.

File: frontend/src/pages/MetricApplication/index.tsx (hunk 21 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/MetricApplication/index.tsx:31]: Sure

File: frontend/src/store/reducers/global.ts (hunk 20 above)
Comment (Rajat-Dabade) [reply frontend/src/store/reducers/global.ts:15]: Okay

File: frontend/src/container/MetricsApplication/Tabs/Overview.tsx (hunk 24 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/Overview.tsx]: Okay, now I understood why you wanted to implement `enabled`. Cool added enable in it.

File: frontend/src/pages/MetricApplication/index.tsx (hunk 22 above)
Comment (palashgdev) [reply frontend/src/pages/MetricApplication/index.tsx:36]: default case is already handled by `||`

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx [hunk 25]
```diff
@@ -78,7 +80,7 @@ function FullView({
 
 	const isLoading = response.isLoading === true;
 
-	if (isLoading) {
+	if (isLoading || isDependedDataLoaded) {
```
Comment (palashgdev) [inline frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx:83]: this or condition is not needed

File: frontend/src/hooks/queryBuilder/useGetQueryRange.ts [hunk 26]
```diff
@@ -11,9 +11,14 @@ import { MetricRangePayloadProps } from 'types/api/metrics/getQueryRange';
...
 	requestData: GetQueryResultsProps,
 	options?: UseQueryOptions<SuccessResponse<MetricRangePayloadProps>, Error>,
+	isDependedDataLoaded?: boolean,
 ) => UseQueryResult<SuccessResponse<MetricRangePayloadProps>, Error>;
 
-export const useGetQueryRange: UseGetQueryRange = (requestData, options) => {
+export const useGetQueryRange: UseGetQueryRange = (
+	requestData,
+	options,
+	isDependedDataLoaded = false,
```
Comment (palashgdev) [inline frontend/src/hooks/queryBuilder/useGetQueryRange.ts:20]: also can we use `isQueryEnabled` seems naming is action driven rather than purpose driven

File: frontend/src/container/MetricsApplication/Tabs/constant.ts [hunk 27]
```diff
@@ -1,3 +1,5 @@
 export const legend = {
 	address: '{{address}}',
 };
+
+export const ERROR_MESSAGE = 'Something went wrong';
```
Comment (palashgdev) [inline frontend/src/container/MetricsApplication/Tabs/constant.ts:5]: can we put this one under 

```
src/constants
```

File: frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx (hunk 25 above)
Comment (palashgdev) [inline frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx:83]: this or condition is not needed

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx (hunk 25 above)
Comment (Rajat-Dabade) [reply frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx:83]: This is required as at the initial load it will show no data rather than loading.

File: frontend/src/container/MetricsApplication/Tabs/constant.ts (hunk 27 above)
Comment (Rajat-Dabade) [reply frontend/src/container/MetricsApplication/Tabs/constant.ts:5]: Okay

File: frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx (hunk 25 above)
Comment (Rajat-Dabade) [reply frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx:83]: This is required as at the initial load it will show no data rather than loading.

File: frontend/src/pages/MetricApplication/index.tsx (hunk 22 above)
Comment (Rajat-Dabade) [reply frontend/src/pages/MetricApplication/index.tsx:36]: Okay

File: frontend/src/hooks/queryBuilder/useGetQueryRange.ts (hunk 26 above)
Comment (Rajat-Dabade) [reply frontend/src/hooks/queryBuilder/useGetQueryRange.ts]: Okay

File: frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx [hunk 28]
```diff
@@ -62,6 +63,7 @@ function FullView({
 		{
 			queryKey,
 		},
+		isDependedDataLoaded,
```
Comment (palashgdev) [inline frontend/src/container/GridGraphLayout/Graph/FullView/index.metricsBuilder.tsx:66]: this variable is not needed

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Review by palashgdev — DISMISSED** [review]
